source/references/troubleshooting.rst

```diff
@@ -167,48 +167,92 @@ can increase the amount of memory available to the containers.
 
     Advanced preferences in Docker for OS X.
 
-Error Messages
-^^^^^^^^^^^^^^
-*Error*: ``KeyError: 'getpwuid()' uid not found: '``
+Error Messages and Solutions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Error: :code:`KeyError: 'getpwuid()' uid not found`
+----------------------------------------------
+This error occurs when host user IDs (UIDs) do not match the container’s UIDs. The user ID is stored in your :code:`~/.kbsdk.cache`.
+Please clear out the cache with the command :code:`rm -rf ~/.kbsdk.cache` and try again. Otherwise try the following:
+
+Solution: Modify the UID in the test scripts
+""""""""""""""""""""""""""""""""""""""""""""
+
+1. Modify the following scripts in the `test_local` directory:
+
+- `run_tests.sh`
+- `run_bash.sh`
+- `run_subjobs.sh`
+
+2. Change the user flag to :code:`--user 0` to run as root, which should bypass the UID issues:
+
+- Example, change :code:`./run_tests.sh --user $(id -u)` to :code:`./run_tests.sh --user 0`.
+
+3. If the problem persists, consider removing the :code:`--user` flag altogether.
+
+
+Error: :code:`Can't find image [test/<your_module_name>:latest]`
+-----------------------------------------------------------------
+
+.. code-block:: bash
 
-*Solution*: Try changing the user flag in ``run_tests.sh``, ``run_bash.sh``, and ``run_subjobs.sh`` (if available) in the ``test_local`` directory to ``--user 0``. Alternatively, remove that flag altogether.
+    Error: "Can't find image [test/<your_module_name>:latest]
+    Here is 'docker images' output: Cannot connect to the Docker daemon.
+    Is the docker daemon running on this host?
 
-*Error*: ``Can't find image [test/<your_module_name>:latest].`` 
-``Here is 'docker images' output: Cannot connect to the Docker daemon.``
-``Is the docker daemon running on this host?``
+This error indicates that Docker is either not running or not reachable, which prevents Docker commands from executing properly.
 
-*Solution*: Run the following code snippet `docker run -it -v /var/run/docker.sock:/run/docker.sock alpine chmod g+w /run/docker.sock`
+Solution: Fix Docker Daemon Socket Permissions
+""""""""""""""""""""""""""""""""""""""""""""""
 
+1. Ensure that the Docker daemon is running on your host.
 
+2. Modify the permissions of the Docker socket to allow group write access, which should resolve connection issues
+:code:`docker run -it -v /var/run/docker.sock:/var/run/docker.sock alpine chmod g+w /var/run/docker.sock`
 
+3. Clear your `kb_sdk` cache with the command :code:`rm -rf ~/.kbsdk.cache` and try running your `kb-sdk` command again.
 
+Error: :code:`Error response from daemon: client version 1.23 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version`
+---------------------------------------------------------------------------------------------------------------------------------------------------------------
```

Comment:
This heading is very long and on taller screens the end is unreadable. I suggest putting just `Error response from daemon: {"message":"client version 1.23 is too old.` as the header and have the full error below in a code block similar to:
```python
Error response from daemon: {"message":"client version 1.23 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version"}
```